app/models/calculator_resolver.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class CalculatorResolver
+  def self.call(calculator)
+    dependencies = {}
+    get_fields(calculator).each do |field|
+      unless field[:type] == 'Calculation'
+        dependencies[field] = []
+        next
+      end
+      dependencies[field] = get_dependencies(field[:value])
+    end
+
+    dependencies
+  end
+
+  def self.get_dependencies(value)
+    fetched_values = []
+    calculator = Dentaku::Calculator.new
+    selectors = calculator.dependencies(value)
+    selectors.each do |selector|
+      fetched_values << Value.find_by(selector: selector.upcase)
+    end
```

Comment:
Build sql query 'where selector in ...' using active record instead

Reply:
Thanks. I've done this.